src/entries/popup/hooks/useTokensShortcuts.ts

```diff
@@ -28,17 +38,80 @@ export function useTokensShortcuts() {
   const navigate = useRainbowNavigate();
   const navigateToSwaps = useNavigateToSwaps();
 
+  const containerRef = useContainerRef();
+
+  const { pinnedAssets, removedPinnedAsset, addPinnedAsset } =
+    usePinnedAssetStore();
+  const { addHiddenAsset } = useHiddenAssetStore();
+
   const hasExplorerLink =
     selectedToken &&
     !isNativeAsset(selectedToken?.address, selectedToken?.chainId);
 
+  const pinned = pinnedAssets.some(({ uniqueId }) => {
+    if (!selectedToken) return false;
+    return uniqueId === selectedToken.uniqueId;
+  });
+
   const allowSwap = useMemo(
     () =>
       (!isWatchingWallet || featureFlags.full_watching_wallets) &&
       config.swaps_enabled,
     [featureFlags.full_watching_wallets, isWatchingWallet],
   );
 
+  const hideToken = useCallback(
+    (_selectedToken: ParsedUserAsset) => {
+      addHiddenAsset({
+        uniqueId: computeUniqueIdForHiddenAsset(_selectedToken),
+      });
+      if (pinned) removedPinnedAsset({ uniqueId: _selectedToken.uniqueId });
+      setSelectedToken();
+      triggerToast({
+        title: i18n.t('token_details.toast.hide_token', {
+          name: _selectedToken.symbol,
+        }),
+      });
+    },
+    [pinned, addHiddenAsset, removedPinnedAsset, setSelectedToken],
+  );
+
+  const togglePinToken = useCallback(
+    (_selectedToken: ParsedUserAsset) => {
+      if (pinned) {
+        removedPinnedAsset({ uniqueId: _selectedToken.uniqueId });
+        triggerToast({
+          title: i18n.t('token_details.toast.unpin_token', {
+            name: _selectedToken.symbol,
+          }),
+        });
+        simulateClick(containerRef.current);
+        return;
+      }
+      addPinnedAsset({ uniqueId: _selectedToken.uniqueId });
+      triggerToast({
+        title: i18n.t('token_details.toast.pin_token', {
+          name: _selectedToken.symbol,
+        }),
+      });
+      simulateClick(containerRef.current);
```

Comment:
are these simulateClick on the container to close the context menu?

Reply:
Yep